packages/core/wallet/src/initialize.ts

```diff
@@ -0,0 +1,41 @@
+import type {
+    Wallet,
+    WalletEventsWindow,
+    WalletInitializeEvent as WalletInitializeEventInterface,
+    WalletInitializeEventCallback,
+} from '@wallet-standard/base';
+
+declare const window: WalletEventsWindow;
+
+export function initializeWallet(wallet: Wallet): void {
```

Comment:
Maybe `registerWallet`? Our idea of ‘initializing’ might be different than theirs. Also, you could theoretically ‘register’ your wallet with multiple consumers, all of whom implement the Wallet Standard. This would be at odds with the notion that ‘initialization’ is something you do once.

Reply:
Ostensibly, initialization _should_ only happen once per wallet. There's no guard against it happening twice, and otherwise they will add two event listeners / dispatch twice. This will be deduped by the Set, though. Open to consider on this, though it doesn't necessarily _just_ register, since it could do different stuff.